integration-tests/helpers/start-server-with-environment.js

```diff
@@ -0,0 +1,24 @@
+const setupServer = require("./setup-server")
+const { initDb } = require("./use-db")
+
+const startServerWithEnvironment = async ({ cwd, verbose, env }) => {
+  if (env) {
+    Object.entries(env).forEach(([key, value]) => {
+      process.env[key] = value
+    })
+  }
```

Comment:
thought: one super tricky thing with this approach is that if people forget to reset `process.env = OLD_ENV` in `afterAll` or something similar; you might end up in some tricky debugging situations. 

Do you think that we could maybe do:
```suggestion
const OLD_ENV = process.env
  if (env) {
    Object.entries(env).forEach(([key, value]) => {
      process.env[key] = value
    })
  }
```

and then after `await initDb` do `process.env = OLD_ENV`? 

This way we are effectively only setting the env vars during the setup of the server and we don't require devs to remember to clean up after their tests.